addons/mail/static/src/models/composer.js

```diff
@@ -140,6 +141,31 @@ registerModel({
                 composerView.update({ hasToRestoreContent: true });
             }
         },
+        async performRPClinkPreview(url) {
+            const attachmentData = await this.env.services.rpc({
+                route: '/mail/link_preview',
+                params: { url, channel_id: this.activeThread.id },
+            }, { shadow: true });
+            if (!attachmentData) {
+                return;
+            }
+            this.update({ attachments: insert(attachmentData) });
+        },
```

Comment:
Would be better to just do the RPC, and handle response in longpolling.
This scales much better. In a previous comment, you would simply reuse the same longpolling notification that contains attachment data.